File: mobile/app/src/main/java/cz/kvalitacena/ui/common/StoreLabel.kt
package cz.kvalitacena.ui.common

import cz.kvalitacena.network.Store

/**
 * Popisek obchodu pro dropdowny a seznamy — název (+ ulice, je-li vyplněná) a město, ve
 * stejném duchu jako web (frontend price-entry). Kód země se přilepí na konec JEN když se liší
 * od [homeCountry] (docs/lokalizace.md, "Country selector v UI") — český obchod se českému
 * uživateli ukáže beze změny, slovenský dostane "(SK)". Webový protějšek: shared/store-label.ts.
 * Čistá funkce mimo Compose, ať jde otestovat JUnitem bez instrumentace (stejný vzor jako
 * PriceChartGeometry).
 */
fun storeLabel(store: Store, homeCountry: String?): String {
  val nameAndStreet = listOfNotNull(
    store.name.trim().takeIf { it.isNotEmpty() },
    store.street?.trim()?.takeIf { it.isNotEmpty() },
  ).joinToString(", ")
  val base = if (nameAndStreet.isEmpty()) store.city else "$nameAndStreet — ${store.city}"
  return if (store.country != homeCountry) "$base (${store.country})" else base
}

File: frontend/src/app/shared/store-label.ts
import { Store } from '../models/catalog';

/**
 * Popisek obchodu v seznamech/pickeru (název — město), s kódem země PŘILEPENÝM na konec jen
 * když se liší od domácí země vieweru ({@link
 * import('../services/country-service').CountryService}) — český obchod se českému uživateli
 * ukáže beze změny, slovenský dostane "(SK)" (docs/lokalizace.md, "Country selector v UI").
 * Mobilní protějšek: mobile ui/common/StoreLabel.kt (`homeCountry` parametr přidán stejně).
 */
export function storeLabel(
  store: Pick<Store, 'name' | 'city' | 'country'>,
  homeCountry: string | null | undefined,
): string {
  const base = `${store.name} — ${store.city}`;
  return store.country && store.country !== homeCountry ? `${base} (${store.country})` : base;
}
